### PR TITLE
documentation and code cleanup

### PR DESCRIPTION
**Pull request by dsyme**
* [x] **fully** document `AbstractSyntax.fs`.  At last this is done. After 18 years I'm happy with how this file is looking now.
* [x] split some very long lines (250char and above)
* [x] some whitespace formatting in FSharp.Core Query.fs
* [x] whitespace formatting in signature files
* [x] move a chunk of code from AbstractSyntaxOps.fs to ParseHelpers.fs

### CODE REVIEW

**Comment by dsyme** [comment]
This is ready (assuming it goes green)

File: src/fsharp/NicePrint.fs [hunk 1]
```diff
@@ -1446,7 +1467,7 @@ module InfoMemberPrinting =
         layout ^^ RightL.rightParen ^^ WordL.colon ^^ PrintTypes.layoutType denv retTy
 
 
-    // Prettify this baby
+    // Prettify an ILMethInfo
```
Comment (KevinRansom) [inline src/fsharp/NicePrint.fs]: I think this was a much better comment:

// Prettify this baby

File: src/fsharp/ParseHelpers.fs [hunk 2]
```diff
@@ -181,3 +184,51 @@ type LexerWhitespaceContinuation =
 
 and LexCont = LexerWhitespaceContinuation
 
+//------------------------------------------------------------------------
```
Comment (KevinRansom) [inline src/fsharp/ParseHelpers.fs]: This is wierd, everywhere else there were ------------ you removed them, yet here you didn't.

Comment (dsyme) [reply src/fsharp/ParseHelpers.fs]: Yes they should go there too.

**Pull request merged**